spec/blumquist_spec.rb

```diff
@@ -104,7 +104,22 @@
       end
 
     end
+  end
+
+  context 'object serialization' do
+    let(:support) { File.expand_path("../support", __FILE__) }
+    let(:schema) { JSON.parse(open(File.join(support, 'schema.json')).read) }
+    let(:data) { JSON.parse(open(File.join(support, 'data.json')).read) }
+    let(:b) { Blumquist.new(schema: schema, data: data) }
 
+    it 'serializes the object to binary' do
+      expect { Marshal.dump(b) }.to_not raise_error
+    end
 
+    it 'deserializes the binary string to a blumquist object' do
+      binary = Marshal.dump(b)
+      loaded_blumquist = Marshal.load(binary)
+      expect(loaded_blumquist.to_s.gsub(/Blumquist:([^\s]+)/, '')).to eq(b.to_s.gsub(/Blumquist:([^\s]+)/, ''))
```

Comment:
Well I wanted a way to compare the entire content of both objects but since I couldn't find any rspec matchers which were able to disregard all the object identities (including the nested objects) I just called `.to_s` and removed the object ID's with gsub :)

Not really pretty I know, but effective ;) The alternative would be to access every nested object individually and compare those attributes. Unless I missed something and you maybe know a better way.